src/framework/core/js/ModelTransformationTransforms.js

```diff
@@ -591,12 +591,46 @@ var fluid = fluid || fluid_2_0;
     };
 
     fluid.defaults("fluid.transforms.indexOf", {
-        gradeNames: "fluid.standardTransformFunction"
+        gradeNames: ["fluid.standardTransformFunction", "fluid.lens"],
+        invertConfiguration: "fluid.transforms.indexOf.invert"
     });
 
     fluid.transforms.indexOf = function (value, transformSpec) {
         var array = fluid.makeArray(transformSpec.array);
-        return array.indexOf(value);
+        var offset = isNaN(Number(transformSpec.offset)) ? 0 : Number(transformSpec.offset);
```

Comment:
Should throw on non-numbers and non-integers rather than being permissive. Converting strings to numbers is fine.